### PR TITLE
Fix TemplatedItemsList incorrect grouped collection range removal

### LINKED ISSUE #7890: [Bug] TemplatedItemsList incorrect grouped collection range removal
### Description
When using grouped ListView, it handles ranged removals wrong.
It happens because of this line:
https://github.com/xamarin/Xamarin.Forms/blob/6ba52f79b51da48386df8f5395fee6b33cc3cff0/Xamarin.Forms.Core/TemplatedItemsList.cs#L813

Items are removed not sequentially, but through one.
### Steps to Reproduce
I'll submit PR several minutes later, you can check it using test

### Expected Behavior
Items are removed sequentially
### Actual Behavior
Items are removed through one
### Basic Information

- Version with issue: 
- Last known good version:
- IDE:
- Platform Target Frameworks: 
  - All
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices:

### PR DESCRIPTION
**Pull request by melimion**
### Description of Change ###

Fixes ranged item removal for grouped ListView

### Issues Resolved ### 

- fixes #7890

### API Changes ###
 
 None

### Platforms Affected ### 

- Core(all platforms)

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### PR Checklist ###

- [x] Targets the correct branch
- [x] Tests are passing (or failures are unrelated)

### CODE REVIEW

**Comment by jfversluis** [comment]
Would you be able to add a UITest for this?

**Comment by melimion** [comment]
@jfversluis 
Added UITest, but you should check it
Not sure it works, I have not much experience with it

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7890.cs [hunk 1]
```diff
@@ -46,6 +59,22 @@ protected override void Init()
 			};
 		}
 
+#if UITEST
+		[Test]
+		public void RefreshingListViewCrashesWhenDisposedTest()
```
Comment (jfversluis) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7890.cs]: I think you copied a bit too much from where you got this ;)

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7890.cs [hunk 2]
```diff
@@ -46,6 +59,22 @@ protected override void Init()
...
+		{
+			foreach (var c in Enumerable.Range(0, Count))
+			{
+				RunningApp.WaitForElement(q => q.Marked(c.ToString()));
+			}
+			RunningApp.Tap(q => q.Button("RemoveBtn"));
+			foreach (var c in Enumerable.Range(RemoveFrom, RemoveCount))
+			{
+				RunningApp.WaitForNoElement(q => q.Marked(c.ToString()));
+			}
```
Comment (jfversluis) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7890.cs]: What do you think this is doing?

I think you should just do a `WaitForElement` on the remove button, then tap that and then verify that the range that you expect to be deleted is actually gone through an Assert statement. I don't think like this the test has the desired effect.

Are you able to run the test locally?

Comment (melimion) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7890.cs]: > I think you copied a bit too much from where you got this ;)

Looks like that )

> Are you able to run the test locally?

After some research and fighting with adb, yes )

I've fixed UI test

**Comment by melimion** [comment]
@rmarinho, I checked replace part, and it looks not affected.

When replacing, old item removed and then new item added within InsertGrouped method, so index is correct.

**Pull request merged**